charts/dapr/crds/components.yaml

```diff
@@ -15,56 +15,65 @@ spec:
         description: Component describes an Dapr component type.
         properties:
           apiVersion:
-            description: 'APIVersion defines the versioned schema of this representation
-              of an object. Servers should convert recognized schemas to the latest
-              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            description: The version of the Dapr (and Kubernetes if applicable) API you are calling.
```

Comment:
The CRDs are auto-generated from what is in `pkg/api` (then you run “make code-generate” and copy from the “staging” files into the charts folder). That’s why they all begin with “APIVersion” etc, because of the Go convention

Reply:
Ah, ok, I see.  Playing with this a bit, I see a couple issues:

- While I suppose *some* description is better than none, the Go convention for docs meant for developers *of* Dapr doesn't translate well into descriptions exposed for developers *usingt* Dapr (e.g. `ComponentSpec is the spec for a component.`).
- Some of the schema properties are "inherited" from k8s common types, so we don't control their descriptions and they are somehow both lengthy *and* vague at the same time (e.g. `Kind is a string value representing the REST resource this object represents. Servers may infer this from the endpoint the client submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'`).

If it were just the first issue, I might argue we bend the Go docs conventions to allow generation of more user-friendly descriptions.  It would be nice if `controller-gen` supported a marker that overrrode the description, but looking at the  source, I'm not sure I can justify the time for my purposes and, given the second issue, it wouldn't completely solve the problem (unless there was a way to override the validation metadata of inherited properties).  Updating descriptions of the generated schema as a second automated step is a possibility; I'd have to look to see what that'd take and whether the time would be justified vs. doing it by hand (e.g. in a copy of the CRD) given the (un)likelyhood of the schema dramatically changing over time.

Reply:
For the first issue, I think we can maybe ignore the Go convention here? We'd need to add the proper `//nolint` tags or the linter would complain too much.

For the "inherited" properties, I don't think there's much we can do. I also don't know if we need to worry about those? Not sure what your use case is.

> Updating descriptions of the generated schema as a second automated step is a possibility; I'd have to look to see what that'd take and whether the time would be justified vs. doing it by hand (e.g. in a copy of the CRD) given the (un)likelyhood of the schema dramatically changing over time.

As for this,please make sure it's automated. We cannot rely on OSS contributors not removing that in the future.

Reply:
> Not sure what your use case is.

The use case is using this schema information to help developers writing component definition YAML in an editor (e.g. provide descriptions of properties on hover, suggest properties, warn when properties are omitted or invalid, etc.).

> As for this,please make sure it's automated. We cannot rely on OSS contributors not removing that in the future.

If an overlay were done manually, it'd be done separate from this file (e.g. and shipped with the tooling).  If it made sense to automate, the question is whether it'd make sense to do it in this repo or as part of the separate tooling. It'd probably easier to do that as part of the tooling or its build process, but it's theoretically more useful here, being available to all potential users.

Reply:
We have a `.build-tools` folder. You could use that to write some additional tooling. Which would not be bad: our tooling right now puts the generated CRDs in the wrong place too.

Reply:
Yeah, but then I'd have to write Go code or, worse, shell script. :-)